### PR TITLE
Deletable media files that are attached to past events only

### PR DESCRIPTION
**Pull request by deen13**
### Short description

This pull request changes the `deletable` flag of the [media_file.py](https://github.com/digitalfabrik/integreat-cms/blob/7f9085f9b8eb13d6639aaafd1efaad1625868c77/integreat_cms/cms/models/media/media_file.py#L2) in a such a way that it is true when only used by events in the past.

### Resolved issues
- [Media items used in past event should be deletable](https://github.com/digitalfabrik/integreat-cms/issues/2634)

Fixes:  #2634

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Review by JoeyStk — APPROVED** [review]
Thank you!
Love the tests!

**Review by MizukiTemma — CHANGES_REQUESTED** [review]
Thank you for qucik PR 👍 

I just found one case where it is not working as expected: media files that are used only in an non-event translation as **content** can be deleted, although they should be protected from deletion.


1. Upload a picture into the media library
2. Insert it into the "content" of a page (not as icon) and save
3. Go back to the media library and open the side menu for the picture
4. See it's deletable

I guess the cause is that ´content_usages´ is not taken into account.

**Comment by deen13** [comment]
Hey @MizukiTemma, great catch, and thank you for bringing it up! I've addressed that edge case and also included an additional test. Unfortunately, I haven't been able to write a test yet that generates a page with content utilizing a media file.

File: integreat_cms/cms/models/media/media_file.py [hunk 1]
```diff
@@ -259,6 +259,39 @@ def content_usages(self) -> QuerySet:
...
+
+        :return: if a media file is only used in past events
+        """
+        usage_counts = [
+            self.past_event_usages.count(),
+            (len(self.icon_usages) + self.content_usages.count()),
+            self.events.count(),
+        ]
+
+        return all(c == usage_counts[0] for c in usage_counts)
```
Comment (MizukiTemma) [inline integreat_cms/cms/models/media/media_file.py:284]: Sorry, my suggestion introduced another bug/edge case 🙈 
`past_event_usages` finds only events in which the media file is used as icon and does not find use case as content 🙈 

So, I'd suggest to adjust `is_only_used_in_past_events` as below

```suggestion
        past_event_translation_ids = [
            event_translation.id
            for event_translation in EventTranslation.objects.filter(
                event__end__lt=timezone.now().date()
            )
        ]

        return (
            len(self.icon_usages) == self.past_event_usages.count()
            and not self.content_usages.exclude(
                content_type__id=ContentType.objects.get(model="eventtranslation").id,
                object_id__in=past_event_translation_ids,
            ).count()
            > 0
        )
```

if this is applied, `from ..events.event_translation import EventTranslation` and `from django.contrib.contenttypes.models import ContentType` need to be imported

**Review by MizukiTemma — CHANGES_REQUESTED** [review]
🙈

File: tests/cms/models/media/test_media_file.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+            region=region,
+        )
+
+        german_language = Language.objects.create(slug="de", bcp47_tag="de", native_name="Deutsch", english_name="German", text_direction="ltr", primary_country_code="DE", table_of_contents="Inhaltsverzeichnis")
+
+        EventTranslation.objects.create(event=past_event, language=german_language, content=f'<p><img src="http://localhost:8000/media/{file.file}"></p>')
+
+        find_all_links()
+
+        assert file.past_event_usages.count() == 1 and file.is_deletable
```
Comment (MizukiTemma) [inline tests/cms/models/media/test_media_file.py:248]:
```suggestion
        file.is_deletable
```
This fails because `past_event_usages` only detects usage as icon, and the media file is used as content in this test. We can omit this in my opinion.

File: tests/cms/models/media/test_media_file.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            is_hidden=False,
+        )
+
+        past_event = Event.objects.create(
+            start=timezone.now() - timedelta(days=2),
+            end=timezone.now() - timedelta(days=1),
+            region=region,
+        )
+
+        german_language = Language.objects.create(slug="de", bcp47_tag="de", native_name="Deutsch", english_name="German", text_direction="ltr", primary_country_code="DE", table_of_contents="Inhaltsverzeichnis")
```
Comment (MizukiTemma) [inline tests/cms/models/media/test_media_file.py:242]: I guess we are getting an error here because German exists already in test data.
We could either load the fixture and use the german language objects from it or create a brand new language with a slug which is nowhere used.

**Review by MizukiTemma — CHANGES_REQUESTED** [review]

File: tests/cms/models/media/test_media_file.py (hunk 2 above)
Comment (deen13) [reply tests/cms/models/media/test_media_file.py]: Yes this was for me to verify that it counts correctly and is deletable but I now understand the behavior better and realized that it is absent because of the linkchecker.

**Review by MizukiTemma — APPROVED** [review]
Thank you!

**Comment by david-venhoff** [comment]
We try to keep the commit history clean, so please remember to squash your commits next time :)
